addon/PlacesProvider.js

```diff
@@ -391,7 +405,8 @@ Links.prototype = {
                       WHERE hidden = 0 AND last_visit_date NOTNULL
                       AND moz_places.url NOT IN (${blockedURLs})
                       ORDER BY frecency, last_visit_date, moz_places.url DESC
-                    ) GROUP BY rev_host)
+                    ) GROUP BY rev_host
+                      ORDER BY frecency, last_visit_date, url DESC)
```

Comment:
Agreed. I have copied this comment to the function so that other folks know the game plan :)